packages/auth/src/core/strategies/email_and_password.test.ts

```diff
@@ -887,53 +949,102 @@ describe('core/strategies/email_and_password/createUserWithEmailAndPassword', ()
         );
 
-        auth.tenantId = TEST_REQUIRE_NUMERIC_TENANT_ID;
+        auth.tenantId = TEST_TENANT_ID_REQUIRE_NUMERIC;
         await expect(
-          createUserWithEmailAndPassword(auth, 'some-email', 'some-password')
+          createUserWithEmailAndPassword(auth, TEST_EMAIL, TEST_PASSWORD)
         ).to.be.rejectedWith(FirebaseError, PASSWORD_ERROR_MSG);
         expect(policyEndpointMockWithOtherTenant.calls.length).to.eq(0);
         expect(auth._getPasswordPolicyInternal()).to.be.undefined;
       });
     });
   });
-});
 
-describe('core/strategies/email_and_password/signInWithEmailAndPassword', () => {
-  let auth: TestAuth;
-  const serverUser: APIUserInfo = {
-    localId: 'local-id'
-  };
+  context('#confirmPasswordReset', () => {
+    const TEST_OOB_CODE = 'oob-code';
 
-  beforeEach(async () => {
-    auth = await testAuth();
-    mockFetch.setUp();
-    mockEndpoint(Endpoint.SIGN_IN_WITH_PASSWORD, {
-      idToken: 'id-token',
-      refreshToken: 'refresh-token',
-      expiresIn: '1234',
-      localId: serverUser.localId!
+    beforeEach(() => {
+      mockEndpoint(Endpoint.RESET_PASSWORD, {
+        email: TEST_EMAIL
+      });
     });
-    mockEndpoint(Endpoint.GET_ACCOUNT_INFO, {
-      users: [serverUser]
+
+    it('does not update the cached password policy upon successful password reset when there is no existing policy cache', async () => {
+      await expect(confirmPasswordReset(auth, TEST_OOB_CODE, TEST_PASSWORD)).to
+        .be.fulfilled;
+
+      expect(policyEndpointMock.calls.length).to.eq(0);
+      expect(auth._getPasswordPolicyInternal()).to.be.null;
     });
-  });
-  afterEach(mockFetch.tearDown);
 
-  it('should sign in the user', async () => {
-    const { _tokenResponse, user, operationType } =
-      (await signInWithEmailAndPassword(
-        auth,
-        'some-email',
-        'some-password'
-      )) as UserCredentialInternal;
-    expect(_tokenResponse).to.eql({
-      idToken: 'id-token',
-      refreshToken: 'refresh-token',
-      expiresIn: '1234',
-      localId: serverUser.localId!
+    it('does not update the cached password policy upon successful password reset when there is an existing policy cache', async () => {
+      await auth._updatePasswordPolicy();
+
+      await expect(confirmPasswordReset(auth, TEST_OOB_CODE, TEST_PASSWORD)).to
+        .be.fulfilled;
+
+      expect(policyEndpointMock.calls.length).to.eq(1);
+      expect(auth._getPasswordPolicyInternal()).to.eql(CACHED_PASSWORD_POLICY);
+    });
+
+    context('handles password validation errors', () => {
+      beforeEach(() => {
+        mockEndpoint(
+          Endpoint.RESET_PASSWORD,
+          {
+            error: {
+              code: 400,
+              message: ServerError.PASSWORD_DOES_NOT_MEET_REQUIREMENTS
+            }
+          },
+          400
+        );
+      });
+
+      it('updates the cached password policy when password does not meet backend requirements', async () => {
```

Comment:
do we have a similar test for tenants?

Reply:
I'll add one for sign-in and reset password flows.